pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -419,7 +423,7 @@ public Map<String, PinotTaskConfig> getSubtaskConfigs(
   @GET
   @Path("/tasks/subtask/{taskName}/progress")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation("Get progress of specified sub tasks for the given task tracked by worker in memory")
+  @ApiOperation("Get progress of specified sub tasks for the given task tracked by minion worker in memory")
```

Comment:
What if we return the task names and corresponding subtasks from `/tasks/subtask/state/{subTaskState}/progress` and asks ppl to use `/tasks/subtask/{taskName}/progress` to fetch for each progress of the subtask? The existing API allows us to provide the list of subtasks so we can fetch the information with a single call if needed.

Reply:
Thanks for the suggestions!

This should also works. However, it may not be ideal because
1. If we only return the task names and corresponding subtasks from `/tasks/subtask/state/{subTaskState}/progress`, 

(1) Task names and corresponding subtasks may not provide enough debugging information. It's highly possible that users still need to invoke `/tasks/subtask/{taskName}/progress` to get more progress information back.
(2) The response size will be smaller, but Pinot needs to perform (almost) the same computation. The computation itself is not simplified. 
(3) While smaller response will help with data transfer, but I don't see a case that we will invoke this API frequently, so response size may not be a concern. 

Even if we want to reduce response size, I feel we can add a separate API to achieve the goal if it's needed

2.  Since task names and corresponding subtasks from `/tasks/subtask/state/{subTaskState}/progress` may not belong to the same task, users need to make multiple calls to `/tasks/subtask/{taskName}/progress` in order to get process information, which may make operations complex.

Any ideas?

Reply:
I think that it's fine to keep the existing approach given that it's for debugging purpose.. I don't have a strong opinion here.

Reply:
Sounds good. I will the current approach as for now.